src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,102 @@
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
+  protected media: Media
+  private configurationServiceSubscription: Subscription
+  private studioConfiguration: StudioConfiguration | undefined
+  private readonly logger: Logger
+  protected mediaCalculatedDuration: number = 0
+
+  constructor(
+    private readonly actionService: ActionService,
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaStateService: MediaStateService,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('MiniShelfComponent')
+  }
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((studioConfiguration: StudioConfiguration) => {
+      this.studioConfiguration = studioConfiguration
+    })
+    this.updateMedia()
+      .then(this.calculateMediaDuration)
+      .catch(error => this.logger.error(`Failed to update media, error is ${error} .`))
+  }
+
+  private async updateMedia(): Promise<void> {
+    if (!this.segment.metadata?.miniShelfVideoClipFile) return
+
+    const media: Media = await this.mediaStateService.getMedia(this.segment.metadata?.miniShelfVideoClipFile)
+    if (!media) return
+    this.media = media
+  }
+
+  private calculateMediaDuration(): void {
+    if (!this.segment.metadata?.miniShelfVideoClipFile) return
+    if (!this.studioConfiguration) return
+
+    this.mediaCalculatedDuration = Math.max(this.media.duration - this.studioConfiguration.blueprintConfiguration.ServerPostrollDuration, 0)
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if ('segment' in changes) {
+      this.updateMedia()
+        .then(this.calculateMediaDuration)
+        .catch(error => this.logger.error(`Failed to update media, error is ${error} .`))
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.configurationServiceSubscription?.unsubscribe()
+  }
+
+  protected get mediaPreviewUrl(): string {
+    if (!this.studioConfiguration) return ''
+    const url: string = `${this.studioConfiguration.settings.mediaPreviewUrl}/media/thumbnail/${this.segment.metadata?.miniShelfVideoClipFile}`
+    return this.studioConfiguration.settings.mediaPreviewUrl ? url : this.defaultAssetForThumbnail
+  }
+
+  public getSanitizedTitle(): string {
+    return this.segment.name
+      .replace(/<[^>]*>/g, '') // remove html tags
+      .replace(/(\r\n|\n|\r)/gm, ' ') // remove newlines
+      .replace(/_/g, ' ') // replace underscores with spaces;
+      .replace(/&nbsp;/g, ' ') // remove &nbsp;
+      .replace(/\s+/g, ' ') // remove extra spaces
+      .replace(/&amp;/g, '&') // remove &amp;
+      .replace(/&quot;/g, '"') // remove &quot;
+      .replace(/&apos;/g, "'") // remove &apos;
+      .trim() // and finally trim;
```

Comment:
Is this necessary? Have we encountered some of these special characters?

Reply:
I see characters out of the set [letter, digit], and I assume is possible to encounter html breaker.